deltacat/aws/clients.py

```diff
@@ -1,22 +1,52 @@
 import logging
 from functools import lru_cache
+from requests import Session
 from typing import Optional
+from http import HTTPStatus
 
 import boto3
 from boto3.exceptions import ResourceNotExistsError
 from boto3.resources.base import ServiceResource
 from botocore.client import BaseClient
 from botocore.config import Config
+from requests.adapters import HTTPAdapter, Retry, Response
 
 from deltacat import logs
 from deltacat.aws.constants import BOTO_MAX_RETRIES
 
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
 BOTO3_PROFILE_NAME_KWARG_KEY = "boto3_profile_name"
+INSTANCE_METADATA_SERVICE_IPV4_URI = "http://169.254.169.254/latest/meta-data/"  # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+
+
+def block_until_instance_metadata_service_returns_success(
+    total_number_of_retries=10,
+    backoff_factor=1,
+) -> Optional[Response]:
+    # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+    with Session() as session:
+        retries = Retry(
+            total=total_number_of_retries,
+            backoff_factor=backoff_factor,  # A backoff factor to apply between attempts after the second try: {backoff factor} * (2 ** ({number of previous retries}))
+            # For example, if the backoff_factor is 1, then Retry.sleep() will sleep for [1s, 2s, 4s, 8s, …] between retries after the first retry.
+            status_forcelist=[
```

Comment:
Ideally, we should retry on any 5xx and 429 (throttling).

Reply:
429 handled by `HTTPStatus.TOO_MANY_REQUESTS`

Reply:
The other possible 5xx status codes - NOT_IMPLEMENTED, INSUFFICIENT_STORAGE etc., - seem unlikely